public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/helpers.tsx

```diff
@@ -44,18 +47,40 @@ const getFeatureNameFromVisParams = (id, seriesParams) => {
 };
 
 function visAggregationToFormik(value) {
-  return [
-    {
-      label: value.params.field.name,
-      type: 'number',
-    },
-  ];
+  if (Object.values(value.params).length !== 0) {
+    return [
+      {
+        label: value.params?.field?.name,
+        type: value.type,
+      },
+    ];
+  }
+  // for count type of vis, there's no field name in the embeddable schema
+  return [];
 }
 
 function visAggregationQueryToFormik(value, seriesParams) {
-  return {
-    [snakeCase(getFeatureNameFromVisParams(value.id, seriesParams))]: {
-      sum: { field: value.params.field.name },
-    },
-  };
+  if (Object.values(value.params).length !== 0) {
+    return {
+      [snakeCase(getFeatureNameFromVisParams(value.id, seriesParams))]: {
+        [visAggregationTypeToFormik(value)]: {
+          field: value.params?.field?.name,
+        },
+      },
+    };
+  }
+  // for count type of vis, there's no field name in the embeddable schema
+  // return '' as the csutom expression query
+  return '';
+}
+
+function visAggregationTypeToFormik(feature) {
+  const aggType = feature.__type.name;
+  if (AGGREGATION_TYPES.some((type) => type.value === aggType)) {
+    return aggType;
+  }
+  if (aggType === 'count') {
```

Comment:
do we define count as value_count in our code across the board?

Reply:
yes https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/ConfigureModel/utils/constants.tsx#L58